src/Qir/Tests/CMakeLists.txt

```diff
@@ -6,7 +6,7 @@ message(INFO "*** build config: ${CMAKE_BUILD_TYPE}")
 project(qir-tests)
 
 # specify the C++ standard, compiler and other tools
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
(nit) It is suspicious that C++ version (17) is chosen in more than one place (suspicion is like "Are these all the places where the change is needed?").

Reply:
I did a search through the code base for all instances of `set(CMAKE_CXX_STANDARD` and saw that the Native simulator was using C++17 while the Qir Runtime was using 14, so I updated it here in the hopes that it would resolve the issue with control flow guard. So far no luck, such that the BinSkim tool we rely on for binary validation still flags the dlls as invalid. I've filed an issue with the BinSkim team in the hopes that we can find a resolution: https://github.com/microsoft/binskim/issues/365